### ISSUE TITLE
Enhancement: divide report by sections in the web interface

### ISSUE THREAD

**Issue by federicodotta** [GhostManager/Ghostwriter#160]
Hi!

First, thank you for your great tool. The reporting features of GhostWriter are impressive, much better than paid alternatives!

Sometimes, if we need to produce a report that includes for example a web application and a mobile application, we create two different sections, each one with the vulnerabilities of that specific application.

If the two applications relate to different types (es. web and mobile) it is already possible to generate a report with two different sections using Jinja filters, but at the moment it is necessary to put all the vulnerabilities together in the "New report" section of GhostWriter GUI and then divide the issues in the template using filters like these ones:

```
{% for type, f in findings|groupby(“finding_type”) %}
{% for finding in findings|selectattr(“finding_type”,”equalto”,”Web”) %}
```

This approach is quite uncomfortable, because in the GhostWriter GUI it is necessary to put all the issues of the two groups tougher. Furthermore, it is not possible for example to produce a report for two web applications with two different sections containing each the vulnerabilities of a single web application.

It would be great to have the possibility to create different sections of a report and being able to put vulnerabilities in each section directly from GhostWriter GUI.

What do you think?

Thank you for your great job!
Federico

**Comment by chrismaddalena** [comment]
Yep! We're tracking this on our road map. Eventually, you will be able to add sections to a report so that an entire report can be constructed within Ghostwriter. It will then be possible to make a report template that is mostly just page breaks and Jinja2 code.

With that feature, I could see being able to add additional findings sections. That would require some heavy lifting because of how findings are currently linked in the database. Also, we'd need to handle situations where you might want a finding in two or more lists without needing to clone it and edit several copies for updates.

Ghostwriter has a few custom Jinja2 filters that simplify the Jinja2 code. There is a `filter_type` that makes your code a bit easier to write.

https://www.ghostwriter.wiki/features/reporting/report-templates/word-template-variables#ghostwriter-filters

We might be able to do something like adding the ability to split findings by type so the lists are pre-made and available. Meaning you could have `findings` (the full list) and several more like `mobile_findings` and `web_findings` with just those.

We also have tagging on the road map. Tagging will enable many possibilities. For example, you could tag specific findings with an ATT&CK ID, the name of an application, or whatever and then filter a list of findings by tag.

None of this directly addresses your request to be able to have multiple lists of findings that appear separate within a single report. It might be possible to allow you to insert a separator row and start a new list of severity groups. It'll require investigation, but I've added it to our public Trello.

**Comment by chrismaddalena** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Filter Findings by Category for Reporting Engine](https://trello.com/c/5icET1ag/46-filter-findings-by-category-for-reporting-engine)

**Comment by federicodotta** [comment]
Perfect! Thank you very much @chrismaddalena! I will follow the Trello!

Federico

**Comment by chrismaddalena** [comment]
We included custom report sections in v4.1 so I'm marking this as closed 😄
